src/components/EditCoursePage/CollapsibleCourseRun.jsx

```diff
@@ -0,0 +1,367 @@
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Field, FieldArray } from 'redux-form';
+import { Collapsible } from '@edx/paragon';
+
+import ActionButton from '../ActionButton';
+import ButtonToolbar from '../ButtonToolbar';
+import FieldLabel from '../FieldLabel';
+import Pill from '../Pill';
+import RenderInputTextField from '../RenderInputTextField';
+import RenderSelectField from '../RenderSelectField';
+import StaffList from '../StaffList';
+import TranscriptLanguage from './TranscriptLanguage';
+
+import { IN_REVIEW_STATUS, PUBLISHED } from '../../data/constants';
+import { endDateHelp, startDateHelp } from '../../helpText';
+
+
+const formatCourseRunTitle = (courseRun) => {
+  if (courseRun) {
+    const labelItems = [];
+    if (courseRun.start) {
+      labelItems.push(moment.utc(courseRun.start).format('MMM Do YYYY'));
+    }
+    if (courseRun.pacing_type) {
+      labelItems.push(courseRun.pacing_type.split('_').map(pacingType =>
+        pacingType.charAt(0).toUpperCase() + pacingType.slice(1)).join(' '));
+    }
+    return (
+      <div className="course-run-label">
+        <span>{`Course run starting on ${labelItems.join(' - ')}`}</span>
+        {/*
+          TODO: After we have a way of determining if the course run has been edited, that should
+          be added into the list of statuses being passed into the Pill component.
+        */}
+        <Pill statuses={[courseRun.status]} />
+        <div className="course-run-studio-url">
+          {'Studio URL - '}
+          <a href={`${process.env.STUDIO_BASE_URL}/course/${courseRun.key}`}>
+            {`${courseRun.key}`}
+          </a>
+        </div>
+        {courseRun.marketing_url ?
+          <div className="course-run-preview-url">
+            {'Preview URL - '}
+            <a href={`${courseRun.marketing_url}`}>
+              {'View about page'}
+            </a>
+          </div> : null}
+      </div>
+    );
+  }
+  return (
+    <div>
+      <span>Your new course run</span>
+    </div>
+  );
+};
+
+const getDateString = date => (date ? moment.utc(date).format('YYYY-MM-DD') : '');
+
+class CollapsibleCourseRun extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      open: false,
+    };
+
+    this.openCollapsible = this.openCollapsible.bind(this);
+    this.setCollapsible = this.setCollapsible.bind(this);
+  }
+
+  setCollapsible(open) {
```

Comment:
Is setCollapsible a thing because we want to stay open to the idea of being able to pass in false?

Why not just have openCollapsible and closeCollapsible?

Reply:
It's written that way because it's easier to have it just be the onToggle handler - which takes `false` or `true` - we are trying to stay in sync with the internal state of the collapsible.